third_party/terraform/resources/resource_container_node_pool.go.erb

```diff
@@ -416,7 +403,12 @@ func resourceContainerNodePoolDelete(d *schema.ResourceData, meta interface{}) e
 
 	_, err = containerNodePoolAwaitRestingState(config, nodePoolInfo.fullyQualifiedName(name), nodePoolInfo.project, d.Timeout(schema.TimeoutDelete))
 	if err != nil {
-		return err
+		if isGoogleApiErrorWithCode(err, 404) {
```

Comment:
It isn't clear to me why this appears as a diff as part of this PR.

Reply:
From the looks of it, it came while trying to make all the tests pass - I'll pull this into a separate PR.

Reply:
@rileykarson - i added this back in, since it was making `TestAccContainerNodePool_withInvalidUpgradeSettings` fail. I couldn't get a concrete reason why it changed with sdkv2, other than the old test framework hid some errors it shouldn't have.
The issue here is that, with the invalid upgrade settings, the node pool doesn't get created, then we try to delete it and we get an error.  I think we don't need an error during delete if it doesn't exist, right? Please feel free to correct me if I'm wrong though, if we want it to error here, I think we'll need to update the test to expect an error.

Reply:
Huh, got it. Before I don't think we'd expect to enter delete if the node pool doesn't exist. Can you comment inline with why we're checking for a 404 here?